tests/ui/async-await/debug-ice-attempted-to-add-with-overflow.rs

```diff
@@ -0,0 +1,10 @@
+// check-fail
+// edition:2021
+
+// test for issue-114912 - debug ice: attempted to add with overflow
+
+async fn main() {
```

Comment:
```suggestion
fn main() {
```

We do not support async main so we can remote it? of there is any reason that I am missing to add it?

Reply:
Ah I see, ok it is fine for me.

another way to write the same thing is 

```rust 
fn main() {
   let _ = async { 
      [0usize; 0xffff_ffff_ffff_ffff].await;
   };
}
```

In this way, you have no longer the problem of the `async fn main`

Reply:
Would you prefer it this way, or okay with how it is now? Thanks for the tip :)

Reply:
>Would you prefer it this way, or okay with how it is now? Thanks for the tip :)

for me, it is fine either way, with the second way you avoid just to generate a second error

error[E0752]: `main` function is not allowed to be `async`

but maybe a t-compiler member will have some preference, for me there is no problem :)

Reply:
Awesome, thanks mate.